docs/stories/Inputs/Checkbox.stories.tsx

```diff
@@ -1,37 +1,53 @@
 import * as React from 'react';
 
+import { useArgs } from '@storybook/preview-api';
 import { Meta, StoryObj } from '@storybook/react';
-import { Checkbox } from '@strapi/design-system';
+import { Checkbox, Field, FieldHint, FieldError, Button } from '@strapi/design-system';
+import { default as outdent } from 'outdent';
 
 const meta: Meta<typeof Checkbox> = {
-  title: 'Design System/Components/Checkbox',
+  title: 'Design System/Inputs/Checkbox',
   component: Checkbox,
 };
-
 export default meta;
 
 type Story = StoryObj<typeof Checkbox>;
 
-export const Base = {
-  render: () => {
-    const [val, setValue] = React.useState(false);
+const Template: Story = {
+  render: ({ children, checked, ...props }) => {
+    const [, updateArgs] = useArgs();
+
+    const handleChange = () => {
+      updateArgs({ checked: !checked });
+    };
+
     return (
-      <Checkbox name="default" onValueChange={(value) => setValue(value)} value={val}>
-        Label
+      <Checkbox {...props} value={checked} onChange={handleChange}>
+        {children}
       </Checkbox>
     );
   },
+};
 
-  name: 'base',
+export const Base = {
+  ...Template,
+  args: {
+    children: 'Label',
+    checked: false,
+  },
 
   parameters: {
     docs: {
       source: {
-        code: '<Checkbox>Label</Checkbox>',
+        code: outdent`
+        <Checkbox value={checked} onChange={handleChange}>
+          {label}
+        </Checkbox>`,
       },
     },
   },
-} satisfies Story;
+  name: 'base',
+};
 
 export const Indeterminate = {
```

Comment:
I'd rather rename this story to be something like "CheckboxGroup" and show a really detailed complicated example that shows nesting:

```
- [x] sweet
- - [x] chocolate
- - [x] haribo 
- [ ] fruit
- - [ ] banana
- - [ ] mango
- [/] vegetables
- - [x] courgette
- - [ ] celery
```